FXyz-Samples/src/main/java/org/fxyz3d/controls/TimelineControl.java

```diff
@@ -0,0 +1,181 @@
+                endTxt.textProperty().unbind();
+                bar.prefWidthProperty().unbind();
+                old.currentRateProperty().removeListener(rateListener);
+            }
+            if (t == null) {
+                timelineDisplay.setVisible(false);
+                startBtn.setDisable(true);
+                rwBtn.setDisable(true);
+                playBtn.setDisable(true);
+                ffBtn.setDisable(true);
+                endBtn.setDisable(true);
+                loopBtn.setDisable(true);
+            } else {
+                timelineDisplay.setVisible(true);
+                currentTimeAsPercentage.bind(Bindings.createDoubleBinding(
+                        () ->  t.getCurrentTime().toMillis() / t.getCycleDuration().toMillis(),
+                        t.currentTimeProperty(), t.cycleDurationProperty()));
+                endTxt.textProperty().bind(Bindings.createStringBinding(
+                        () -> String.format("%.2fs", t.getCycleDuration().toSeconds()),
+                        t.cycleDurationProperty()));
+                currentTxt.textProperty().bind(Bindings.createStringBinding(
+                        () -> String.format("%.2fs", t.getCurrentTime().toSeconds()),
+                        t.currentTimeProperty()));
+                bar.prefWidthProperty().bind(Bindings.createDoubleBinding(
+                        () -> progressBar.getWidth() * currentTimeAsPercentage.get(),
+                        currentTimeAsPercentage));
+
+                startBtn.setDisable(false);
+                rwBtn.setDisable(false);
+                playBtn.setDisable(false);
+                ffBtn.setDisable(false);
+                endBtn.setDisable(false);
+                loopBtn.setDisable(false);
+                playBtn.setSelected(t.getCurrentRate() != 0);
+                loopBtn.setSelected(t.getCycleDuration().equals(Timeline.INDEFINITE));
+                t.currentRateProperty().addListener(rateListener);
+            }
+            old = t;
+        }
+    };
+
+    public TimelineControl(Property<Timeline> prop, String name) {
+        super("/org/fxyz3d/controls/TimelineControl.fxml", prop);
+        timeline.bindBidirectional(prop);
+        title.setText(name);
+
+        background.setCache(true);
+
+        startBtn.setOnAction(e -> {
+            Timeline timeline = getTimeline();
+            timeline.jumpTo(Duration.ZERO);
+            timeline.pause();
+        });
+        endBtn.setOnAction(e -> {
+            Timeline timeline = getTimeline();
+            timeline.jumpTo(timeline.getTotalDuration());
+            timeline.pause();
+        });
+        playBtn.setOnAction(e -> {
+            Timeline timeline = getTimeline();
+            if (playBtn.isSelected()) {
+                timeline.play();
+            } else {
+                timeline.pause();
+            }
+        });
+        ffBtn.setOnMousePressed(e -> {
+            Timeline timeline = getTimeline();
```

Comment:
Feel free to ignore this comment: I would have added 

```
private void setTimelineRate(double rate) {
    getTimeline().setRate(rate);
}
```
and then inline all lambdas, e.g.

```
ffBtn.setOnMousePressed(e -> setTimelineRate(2));
```

Reply:
Makes sense